src/helpers.tsx

```diff
@@ -1,29 +1,50 @@
 /* eslint-disable import/prefer-default-export */
+import { logger } from '@storybook/client-logger';
 import { WrapperSettings } from '@storybook/addons';
 import { DEFAULT_PADDING } from './constants';
 
-type Option = {
-  name: string;
-  value: string;
-  default?: boolean;
+type Padding = { name: string; value: string };
+
+type PaddingsParameter = {
+  default?: string;
+  disable?: boolean;
+  values: Padding[];
 }
 
-type Options = Option[] | WrapperSettings['parameters'];
+export type PaddingWithDefault = Padding & Pick<PaddingsParameter, 'default'>;
+
+type Options = Padding[] | PaddingsParameter | WrapperSettings['parameters'];
 
-export const isEnabled = (options: Options): options is Option[] => options.length > 0;
+export const isEnabled = (values: Padding[]): boolean => values.length > 0;
 
-export const getSelectedPadding = (options: Options, currentValue: string): string => {
-  if (!isEnabled(options)) {
-    return DEFAULT_PADDING;
+export const normalizeValues = (options: Options): PaddingWithDefault[] => {
+  if (Array.isArray(options)) {
+    logger.warn(
+      'Using array parameter with Paddings addon is deprecated, and support will be dropped in a future major version. Please refer to the release notes for migration: https://github.com/rbardini/storybook-addon-paddings/releases/tag/v2.1.0',
+    );
+    return options;
   }
 
+  if (options && !options?.disable && options.values.length > 0) {
```

Comment:
We could but I think it would not be appropriate, at this moment the `options.values` is not yet `PaddingWithDefault` and actually `isEnabled` should be using the previous and not `Padding`. It didn't complain because it's optional.